odo/backends/sql.py

```diff
@@ -172,7 +172,9 @@ def discover_typeengine(typ):
     if isinstance(typ, (sa.NUMERIC, sa.DECIMAL)):
         return datashape.Decimal(precision=typ.precision, scale=typ.scale)
     if isinstance(typ, (sa.String, sa.Unicode)):
-        return datashape.String(typ.length, typ.collation)
+        return datashape.String(typ.length, None)
+    if isinstance(typ, sa.dialects.mssql.BIT):
+        return datashape.int_
```

Comment:
I feel like there was a reason that I picked that on bool... I can't seem to recall what that was. 

My real point is that there's a lot of unique MSSQL types. See [here](http://docs.sqlalchemy.org/en/latest/dialects/mssql.html#sql-server-data-types). What I'm wondering if I should make a dict with a bunch of mappings to them like type and revtype (lines 48-87 in backends/sql.py) or is this something that we shoudl send to SQL Alchemy? I'm not really sure where/how this transformation *ought* to take place. I'm happy to write the dict, I just want to do it the right place.